site/en/blog/pop-ups-theyre-making-a-resurgence/index.md

```diff
@@ -0,0 +1,729 @@
+
+There are often two major concerns when building pop-ups:
+
+- How to make sure it gets placed above the rest of your content in an appropriate place.
+- How to make it accessible (keyboard friendly, focusable, and so on).
+
+The built-in pop-up API has a [variety of goals](https://open-ui.org/components/popup.research.explainer#goals), all with the same overarching goal of making it easy for developers to build this pattern. Notable of those goals are:
+
+Make it easy to display an element and its descendants above the rest of the document.
+Make it accessible.
+Not require JavaScript for most common behaviors  (light dismiss, singleton, stacking, and so on).
+You can check out the full spec for pop-ups on the [OpenUI site](https://open-ui.org/components/popup.research.explainer).
+
+## Browser Compatibility
+
+Where can you use the built-in pop-up API now? It's supported in Chrome Canary behind the "Experimental web platform features" flag at the time of writing.
+
+To enable that flag, open Chrome Canary and visit `chrome://flags`. Then enable the "Experimental web platform features" flag.
+
+There is an [Origin Trial](/origintrials/#/view_trial/4500221927649968129) too for developers that would like to test this out in a production environment.
+
+Lastly, there is a [polyfill](https://github.com/oddbird/popup-polyfill) under development for the API. Be sure to check out the repo at [github.com/oddbird/popup-polyfill](https://github.com/oddbird/popup-polyfill).
+
+You can check for pop-up support with:
+
+```js
+const supported = Element.prototype.hasOwnProperty("popUp");
+```
+
+## Current Solutions
+
+What can you currently do to promote your content above everything else? If it's supported in your browser, you could use the [HTML Dialog element](https://developer.mozilla.org/docs/Web/HTML/Element/dialog). You'd need to use it in "Modal" form. And this requires JavaScript to use.
+
+```js
+Dialog.showModal();
+```
+
+There are some accessibility considerations. It's advised to use "[a11y-dialog](https://a11y-dialog.netlify.app/)" for example if catering for users of Safari below version 15.4. 
+
+You could also use one of the many pop-up/alert/tooltip based libraries out there. Many of these tend to work in a similar way.
+
+- Append some container to the body for showing popups
+- Style it so that it sits above everything else
+- Create an element and append it to the container to show a pop-up
+- Hide it by removing the pop-up element from the DOM
+
+This requires an extra dependency and more decisions for developers. It also requires research to find an offering that provides everything you need. The pop-up API, although named pop-up, aims to cater for many scenarios including tooltips. The goal being to cover all those common scenarios, saving developers from having to make yet another decision so they can focus on building their experiences.
```

Comment:
Can you decide whether to use popup or pop-up and do so consistently?

Reply:
I'm talking about these two lines:

- Append some container to the body for showing popups
- Create an element and append it to the container to show a pop-up

I think, based on the rest of the document, the first should also be pop-ups.

Reply:
Good spot! Fixed. I feel like I've written popups/popup/pop-up/pop-ups about a million times now. The decision on the way to address pop-up changed during the writing process.